app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/NoCollectionsMessageViewHolder.kt

```diff
@@ -5,59 +5,210 @@
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SectionHeader
+import org.mozilla.fenix.compose.dashedBorder
 import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
-import org.mozilla.fenix.utils.view.ViewHolder
+import org.mozilla.fenix.theme.FirefoxTheme
 
-@OptIn(ExperimentalCoroutinesApi::class)
-open class NoCollectionsMessageViewHolder(
-    view: View,
+/**
+ * [RecyclerView.ViewHolder] for displaying a message detailing the collections feature and
+ * allowing users to easily start creating their first.
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ * @param store to receive live updates to normal tabs.
+ * @param interactor [CollectionInteractor] callback for user interaction.
+ */
+
+class NoCollectionsMessageViewHolder(
+    composeView: ComposeView,
     viewLifecycleOwner: LifecycleOwner,
-    store: BrowserStore,
-    interactor: CollectionInteractor
-) : ViewHolder(view) {
+    private val store: BrowserStore,
+    private var interactor: CollectionInteractor
+) : ComposeViewHolder(composeView, viewLifecycleOwner) {
 
     init {
-        val binding = NoCollectionsMessageBinding.bind(view)
+        val horizontalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_item_horizontal_margin)
+        val verticalPadding =
+            composeView.resources.getDimensionPixelSize(R.dimen.home_fragment_top_toolbar_header_margin)
+        composeView.setPadding(horizontalPadding, verticalPadding, horizontalPadding, 0)
+    }
 
-        binding.addTabsToCollectionsButton.apply {
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+    }
 
-            setOnClickListener {
-                interactor.onAddTabsToCollectionTapped()
-            }
-            isVisible = store.state.normalTabs.isNotEmpty()
-        }
+    @Composable
+    override fun Content() {
+        Spacer(Modifier.height(40.dp))
 
-        binding.removeCollectionPlaceholder.apply {
-            increaseTapArea(
-                view.resources.getDimensionPixelSize(R.dimen.tap_increase_16)
+        NoCollectionsMessage(
+            store,
+            interactor::onAddTabsToCollectionTapped,
+            interactor::onRemoveCollectionsPlaceholder
+        )
+    }
+}
+
+/**
+ * [NoCollectionsMessage] for displaying a message detailing the collections feature and
+ * allowing users to easily start creating their first.
+ *
+ * @param store to receive live updates to normal tabs.
+ * @param onAddTabsToCollectionTapped to create a new first collection.
+ * @param onRemoveCollectionsPlaceholder to remove empty collections message.
+ */
+
+@Composable
+fun NoCollectionsMessage(
+    store: BrowserStore,
```

Comment:
we'll want to move this out and up to `Content()` to keep the Store pattern outside of the Composable, but also allow the previews to work (they don't work the call to `components`.